### DISCUSSION TITLE
hx-boost and css-scope-inline

### DISCUSSION THREAD

**Discussion by svene** [gnat/css-scope-inline#18] [Q&A]
It looks to me like hx-boost (of htmx) and css-scope-inline do not work properly together.

On the first page load everything is fine, but as soon as I click on a link (which causes hx-boost to switch the body) the counter of css-scope-inline is not reset (I understand why) but also styles/classes are not properly applied anymore.

I'm not even sure if this is more of a problem on the hx-boost side or on the css-scope-inline side.
Or maybe the both just should not be used in combination.

Any opinions about this?

**Comment by eboody** [comment]
I noticed the same thing. The when swapping elements if the style block is inside of the element that's being replaced then css-scope-inline's class name doesnt get added to the swapped html element.

to work around this i moved the style block one-level up and adjusted the selectors as needed
